src/main/java/org/spongepowered/common/event/tracking/phase/GeneralPhase.java

```diff
@@ -147,7 +149,14 @@ public GeneralPhase addChild(TrackingPhase child) {
     public void unwind(CauseTracker causeTracker, IPhaseState state, PhaseContext phaseContext) {
         if (state == State.COMMAND) {
             final ICommandSender sender = phaseContext.getSource(ICommandSender.class)
-                    .orElseThrow(TrackingUtil.throwWithContext("Expected to be capturing a Command Sender, but none found!", phaseContext));
+                    // The sender could be a custom CommandSource, so we check for that too
+                    .orElseGet(() -> {
+                        CommandSource source = phaseContext.getSource(CommandSource.class)
+                                .orElseThrow(TrackingUtil.throwWithContext("Expected to be capturing a Command Sender, but none found!",
+                                        phaseContext));
+                        return WrapperICommandSender.of(source);
+                        }
+                    );
```

Comment:
I think these should be on one line, using the indentation of the second line.

`});`